Manito/Manito/Screens/CreateRoom/ViewModel/CreateRoomViewModel.swift

```diff
@@ -106,84 +164,91 @@ final class CreateRoomViewModel: BaseViewModelType {
-                guard let self = self else { return step }
-                return self.previous(step: step)
+        let roomInfo = Publishers.CombineLatest3(self.titleSubject, 
+                                                 self.capacitySubject,
+                                                 self.dateRangeSubject)
+            .map { title, capacity, date in
+                return RoomInfo(title: title, 
+                                capacity: capacity,
+                                dateRange: date)
             }
             .eraseToAnyPublisher()
         
-        return Output(title: self.titleSubject,
+        return Output(currentStep: currentStep,
+                      counts: mergeCount,
                       fixedTitleByMaxCount: fixedTitle,
-                      capacity: self.capacitySubject,
-                      dateRange: self.dateRangeSubject,
+                      capacity: self.capacitySubject.eraseToAnyPublisher(),
                       isEnabled: isEnabled,
-                      currentNextStep: currentNextStep, previousStep: previousStep,
-                      roomId: self.roomIdSubject)
+                      roomId: self.roomIdSubject.eraseToAnyPublisher(), 
+                      roomInfo: roomInfo)
     }
     
     // MARK: - init
     
-    init(createRoomService: CreateRoomService) {
-        self.createRoomService = createRoomService
+    init(usecase: CreateRoomUsecase) {
+        self.usecase = usecase
     }
     
     // MARK: - func
     
+    private func cutTextByMaxCount(text: String, maxCount: Int) -> String {
+        let isOverMaxCount = self.isOverMaxCount(titleCount: text.count, maxCount: maxCount)
+        
+        if isOverMaxCount {
+            let endIndex = text.index(text.startIndex, offsetBy: maxCount)
+            let fixedText = text[text.startIndex..<endIndex]
+            return String(fixedText)
+        }
+        return text
+    }
+    
     private func isOverMaxCount(titleCount: Int, maxCount: Int) -> Bool {
-        if titleCount > maxCount { return true }
-        else { return false }
+        return titleCount > maxCount ? true : false
     }
     
-    private func runActionByStep(step: CreateRoomStep) -> CurrentNextStep {
-        switch step {
-        case .chooseCharacter:
-            self.requestCreateRoom(roomInfo: CreatedRoomInfoRequestDTO(title: self.titleSubject.value,
-                                                                       capacity: self.capacitySubject.value,
-                                                                       startDate: self.startDateSubject.value,
-                                                                       endDate: self.endDateSubject.value))
-            return (step, step.next())
+    private func initStep() -> StepButtonState {
+        return (self.currentStep, false)
+    }
+    
+    private func nextStep() -> StepButtonState {
+        switch currentStep {
```

Comment:
p1;

여기 `self`가 빠져있군요!
```suggestion
        switch self.currentStep {
```

Reply:
진짜 오타랑 컨벤션 관련해서 없앨라고 열심히 봤는데 이걸 놓쳤네요 ㅠㅠ